test/model/store_test.dart

```diff
@@ -972,6 +992,74 @@ void main() {
     });
   });
 
+  group('UpdateMachine.poll reload failure', () {
+    late LoadingTestGlobalStore globalStore;
+
+    List<Completer<PerAccountStore>> completers() =>
+      globalStore.completers[eg.selfAccount.id]!;
+
+    Future<void> prepareReload(FakeAsync async) async {
+      globalStore = LoadingTestGlobalStore(accounts: [eg.selfAccount]);
+      final future = globalStore.perAccount(eg.selfAccount.id);
+      completers()
+        ..single.complete(
+            eg.store(globalStore: globalStore, account: eg.selfAccount))
+        ..clear();
+      final store = await future;
```

Comment:
nit: I think this is a bit clearer if it's made explicit that the `store` we end up with is the same one created with that `eg.store` call:
```suggestion
      final store = eg.store(globalStore: globalStore, account: eg.selfAccount);
      completers().single.complete(store);
      await future;
      completers().clear();
```